moped-editor/src/views/projects/projectView/ProjectComponents/ComponentForm.js

```diff
@@ -132,38 +144,54 @@ const ComponentForm = ({
     !!initialFormValues?.phase || !!initialFormValues?.completionDate
   );
 
-  useResetDependentFieldOnAutocompleteChange({
+  useResetDependentFieldOnParentFieldChange({
     parentValue: watch("phase"),
     dependentFieldName: "subphase",
+    comparisonVariable: "value",
     valueToSet: defaultFormValues.subphase,
     setValue,
   });
 
-  useResetDependentFieldOnAutocompleteChange({
+  useResetDependentFieldOnParentFieldChange({
     parentValue: watch("component"),
     dependentFieldName: "signal",
+    comparisonVariable: "value",
     valueToSet: defaultFormValues.signal,
     setValue,
   });
 
   // todo: preserve subcomponent choices if allowed when switching b/t component types
-  useResetDependentFieldOnAutocompleteChange({
+  useResetDependentFieldOnParentFieldChange({
     parentValue: watch("component"),
     dependentFieldName: "subcomponents",
+    comparisonVariable: "value",
     valueToSet: defaultFormValues.subcomponents,
     setValue,
     disable: isEditingExistingComponent,
   });
 
   // todo: preserve work type if allowed when switching b/t component types
-  useResetDependentFieldOnAutocompleteChange({
+  useResetDependentFieldOnParentFieldChange({
     parentValue: watch("component"),
     dependentFieldName: "work_types",
+    comparisonVariable: "value",
     valueToSet: defaultFormValues.work_types,
     setValue,
     disable: isEditingExistingComponent,
   });
 
+  useResetDependentFieldOnParentFieldChange({
+    parentValue: watch("signal"),
+    dependentFieldName: "locationDescription",
+    comparisonVariable: "properties",
+    valueToSet:
+      signal && locationDescription?.length === 0
+      // if the signal exists and the locationDescription is empty, set to option label
+        ? getSignalOptionLabel(signal)
+        : locationDescription,
+    setValue,
```

Comment:
@mddilley I changed this to prevent the behavior you were describing

what was happening before was the signal existed so it was using the option label as the description instead of whatever location description was being input into the empty field.

Reply:
thanks @chiaberry for reworking this! completely backtracking on what i asked for yesterday, i think we want the location description to always update if the signal changes, or clear/become empty if the component changes from signal to something else.

i'm not following this other problem of pasting text into the field, so hopefully rolling back to the previous behavior is not in conflict with that!

Reply:
this is partially me too! The paste bug that I noticed is definitely an edge case so, if it gets in the way of making this work the way that we want, I say punt. I'll send John a screencap that I have with some other keyboard events that don't work so great with the form so we can address them later.

Reply:
@chiaberry after looking at some of the wonkiness that Mike found, i think there's an issue with the equality check in the updated `useResetDependentFieldOnParentFieldChange` hook

This returning `false` when trying to compare two signal objects:

```javascript
parentValue?.[comparisonVariable] ===  previousParentFormValue?.[comparisonVariable]`
```

I'm still committed to seeing this one hook handle all the field inputs! I think some lodash might make this easier because you could dot chaining and make compare variable `"properties.id"`?

```javascript
get(parentValue, comparisonVariable) ===  get(previousParentFormValue, comparisonVariable)`
```